syllabus/core/stat_recorder.py

```diff
@@ -34,45 +31,96 @@ def record(self, episode_return: float, episode_length: int, episode_task, env_i
+
+        :param writer: Tensorboard summary writer.
         """
-        records = json.dumps(self.records, default=convert_numpy)
-        with open(os.path.join(self.write_path, 'records.json'), "w") as file:
-            file.write(records)
-        stats = json.dumps(self.stats, default=convert_numpy)
-        with open(os.path.join(self.write_path, 'stats.json'), "w") as file:
-            file.write(stats)
+        try:
+            import wandb
+            tasks_to_log = self.tasks
+            if len(self.tasks) > 5 and not log_full_dist:
+                warnings.warn("Only logging stats for 5 tasks.")
+                tasks_to_log = self.tasks[:5]
+            if self.task_names:
+                for idx in tasks_to_log:
+                    if self.stats[idx]:
+                        writer.add_scalar(f"stats_per_task/task_{self.task_space.task_name(idx)}_episode_return_mean", self.stats[idx]['mean_r'], step)
+                        writer.add_scalar(f"stats_per_task/task_{self.task_space.task_name(idx)}_episode_return_var", self.stats[idx]['var_r'], step)
+                        writer.add_scalar(f"stats_per_task/task_{self.task_space.task_name(idx)}_episode_length_mean", self.stats[idx]['mean_l'], step)
+                        writer.add_scalar(f"stats_per_task/task_{self.task_space.task_name(idx)}_episode_length_var", self.stats[idx]['var_l'], step)
+                    else:
+                        writer.add_scalar(f"stats_per_task/task_{self.task_space.task_name(idx)}_episode_return_mean", 0, step)
+                        writer.add_scalar(f"stats_per_task/task_{self.task_space.task_name(idx)}_episode_return_var", 0, step)
+                        writer.add_scalar(f"stats_per_task/task_{self.task_space.task_name(idx)}_episode_length_mean", 0, step)
+                        writer.add_scalar(f"stats_per_task/task_{self.task_space.task_name(idx)}_episode_length_var", 0, step)
+            else:
+                for idx in tasks_to_log:
+                    if self.stats[idx]:
+                        writer.add_scalar(f"stats_per_task/task_{idx}_episode_return_mean", self.stats[idx]['mean_r'], step)
+                        writer.add_scalar(f"stats_per_task/task_{idx}_episode_return_var", self.stats[idx]['var_r'], step)
+                        writer.add_scalar(f"stats_per_task/task_{idx}_episode_length_mean", self.stats[idx]['mean_l'], step)
+                        writer.add_scalar(f"stats_per_task/task_{idx}_episode_length_var", self.stats[idx]['var_l'], step)
+                    else:
+                        writer.add_scalar(f"stats_per_task/task_{idx}_episode_return_mean", 0, step)
+                        writer.add_scalar(f"stats_per_task/task_{idx}_episode_return_var", 0, step)
+                        writer.add_scalar(f"stats_per_task/task_{idx}_episode_length_mean", 0, step)
+                        writer.add_scalar(f"stats_per_task/task_{idx}_episode_length_var", 0, step)
```

Comment:
Please simplify this code, you shouldn't need all these if statements and repeated code. Also I think you can ignore the task_names feature for now, it's sort of half implemented, I'll need to fix it at some point